engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/error/ThrowErrorNode.java

```diff
@@ -4,14 +4,15 @@
 import com.oracle.truffle.api.nodes.Node;
 import org.enso.interpreter.dsl.BuiltinMethod;
 import org.enso.interpreter.runtime.error.DataflowError;
+import org.enso.interpreter.runtime.state.State;
 
 @BuiltinMethod(
     type = "Error",
     name = "throw",
     description = "Returns a new value error with given payload.",
     inlineable = true)
 public class ThrowErrorNode extends Node {
-  public Object execute(VirtualFrame giveMeAStackFrame, Object payload) {
-    return DataflowError.withoutTrace(payload, this);
+  public Object execute(VirtualFrame giveMeAStackFrame, State state, Object payload) {
+    return DataflowError.withoutTrace(state, payload, this);
```

Comment:
Passing `State` this way might be a temporary solution; there might be another approach I will use when calling `withoutTrace` from Java which makes this unnecessary.

Reply:
How does this `State` argument change modifies the `ThrowErrorNodeGen`?

Reply:
Diff:

```
112a113,114
>     /***  Start of processing argument 0  ***/
>     /***  End of processing argument 0  ***/
127c129
<         result = bodyNode.execute(frame, arg1);
---
>         result = bodyNode.execute(frame, state, arg1);
139c141
<         return bodyNode.execute(frame, arg1);
---
>         return bodyNode.execute(frame, state, arg1);
```

Reply:
Yup. So we can get access to `State` when we have `args` and we extract it from there. This is tedious, especially when we consider
- #7117

It makes more sense to fix #7117  by storing the state in [ContextThreadLocal](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/ContextThreadLocal.html). Then it is going to be easy to access it even when our interpreter code doesn't have access to `args`.